### DISCUSSION TITLE
Helm chart for running Ente?

### DISCUSSION THREAD

**Discussion by kastl-ars** [ente-io/ente#3731] [Enhancements]
Hi all,

I searched the web but found no existing helm chart for self-hosting Ente in Kubernetes. Is someone already working on a chart?

If not, this would be something that I would like to see.

(I am currently trying to grasp the different services that make up ente, e.g. photos and auth and whatnot. If I understand the concept and layout, I might try to come up with a new chart, if there is none...)

Kind Regards,
Johannes

**Comment by sfudeus** [comment]
I just found this one someone has started 2 months ago (https://github.com/jatinkatyal13/ente-helm/), but this seems to not be complete either. 
I'm myself working on running Ente on K8s, I might provide a chart once this is running.

**Comment by ocraviotto** [comment]
I host Ente in Kubernetes, and when getting started with my own setup I could not find a chart that would fit my needs, so I decided [to create my own](https://github.com/ocraviotto/helm-charts/tree/main/charts/ente). 
If others want to try it it and provide feedback, I'll try to improve it based on that (if so, please create an issue in that repository).

Given the way the web and server components are released, I also build the server and web images used with the chart. I do so by running renovate and checking against `.id`  value returned from https://api.ente.io/ping (which is [what the Actions workflow that builds the ghcr does](https://github.com/ente-io/ente/blob/aadacda2f6007e5756a41e8368817cd75f92e4fc/.github/workflows/server-publish-ghcr.yml#L20) to set the docker tag). This means that ever time I detect a change, I build both images (instead of relying on the scheduled [weekly web build](https://github.com/ente-io/ente/blob/aadacda2f6007e5756a41e8368817cd75f92e4fc/.github/workflows/web-publish-ghcr.yml#L4) or [monthly server](https://github.com/ente-io/ente/blob/aadacda2f6007e5756a41e8368817cd75f92e4fc/.github/workflows/server-publish-ghcr.yml#L4) one). 

To ensure renovate automatically updates the chart versions, I use a semver tag for the builds I create instead of the hash, but the source git hash is included in the image metadata. I otherwise use Ente's repository to build them.

When I have a moment I'll create an issue here as I believe that the web and the server should have their own release processes as the other Ente components.
